src/technochat/chat/chat.go

```diff
@@ -54,15 +58,30 @@ func (c *Chat) AddUser(ws *websocket.Conn) *User {
 		WS:   ws,
 		Name: name,
 		ID:   id,
+		send: make(chan WSMessage),
 	}
 	c.corresps[usr.ID] = usr
 	c.restJoins--
+
+	go c.HandleUserSending(usr)
+
 	return usr
 }
 
 func (c *Chat) DelUser(id int) {
+	if c.corresps[id] == nil {
+		return
+	}
+
+	c.corresps[id].terminateSend <- struct{}{}
 	c.corresps[id].WS.Close()
 	delete(c.corresps, id)
+
+	if len(c.corresps) == 0 &&
```

Comment:
better leave on one line

Reply:
actually if 1 of n (n > 1) clients comes and exits, chat may hang forever

Reply:
i d better call wg.Done() here and in some main goroutine wg.Wait() and then terminate chat

Reply:
we dont need wait group logic here. what would you do between chat initialisation and first user got connected? you might need specific logic for first user in this case, to start wg.Wait(). Not enought reasons to use wg.